api/src/wfl/module/covid.clj

```diff
@@ -481,12 +516,13 @@
   Return EXECUTOR."
   [source executor]
   (if-let [snapshot (peek-queue! source)]
-    (let [reference  (from-source executor snapshot)
-          submission (create-submission! executor reference)]
-      (write-workflows! executor reference submission)
-      (pop-queue! source)))
-  (update-terra-workflow-statuses! executor)
-  executor)
+    (let [reference        (from-source executor snapshot)
+          updated-executor (update-method-configuration executor reference)
```

Comment:
If be tempted to move the method configuration update into the create submission function, that way you don't need to juggle the "updated-executor"

Reply:
Thanks for pointing this out.  Since `update-executor!` is meant to return an `executor` to feed into `update-sink!`, I thought that if we're altering the executor's method config version then we should ultimately return that updated executor.  But if we are content with only updating the corresponding DB record, then I'd be very happy to not have to juggle this!

Reply:
Yep - I don't think we should rely on the source/executor object being up-to-date. That way, we can split these updates from sequential to parallel execution.

Reply:
Great!  I will simplify this in my upcoming submission launch PR.

Reply:
Addressed here: https://github.com/broadinstitute/wfl/pull/406